### PR TITLE
Display all errors

### LINKED ISSUE #40: Accumulate all errors before exiting
Find all of the errors for a post-process step before bailing out, to provide a more useful user experience, rather than just latest error, bail out, fix, next error, bail out, fix iterations.

### PR DESCRIPTION
**Pull request by filefrog**
Fixes #40

### CODE REVIEW

**Comment by qanx** [comment]
I really hope that this is merged soon, because it changed my output from this:

```
$ make refresh manifest
Refreshing global data for sandbox
Updated global data
Refreshing site data for sandbox
Updated site data
Building sandbox
Working in /Users/quinn/microbosh-deployments/sandbox
$.cloud.plugin: site does not define cloud.plugin
make: *** [manifest] Error 2

$ make refresh manifest
Refreshing global data for sandbox
Updated global data
Refreshing site data for sandbox
Updated site data
Building sandbox
Working in /Users/quinn/microbosh-deployments/sandbox
$.resources.persistent_disk: site does not define resources.persistent_disk
make: *** [manifest] Error 2

$ make refresh manifest
Refreshing global data for sandbox
Updated global data
Refreshing site data for sandbox
Updated site data
Building sandbox
Working in /Users/quinn/microbosh-deployments/sandbox
$.name: Please specify a name for your micro-bosh deployment
make: *** [manifest] Error 2
```

TLDR seeing a new error in each run, to this:

```
$ make refresh manifest
Refreshing global data for sandbox
Updated global data
Refreshing site data for sandbox
Updated site data
Building sandbox
Working in /Users/quinn/microbosh-deployments/sandbox
9 error(s) detected:
 - $.apply_spec.agent.blobstore.address: site does not define apply_spec.agent.blobstore.address
 - $.apply_spec.agent.nats.address: site does not define apply_spec.agent.nats.address
 - $.apply_spec.properties.hm.director_account: site does not define apply_spec.properties.hm.director_account
 - $.apply_spec.properties.registry.address: site does not define apply_spec.properties.registry.address
 - $.cloud.plugin: site does not define cloud.plugin
 - $.name: Please specify a name for your micro-bosh deployment
 - $.network: site does not define a network!
 - $.resources.cloud_properties: site does not define resources.cloud_properties
 - $.resources.persistent_disk: site does not define resources.persistent_disk


make: *** [manifest] Error 2
```
